File: ppapi/native_client/tests/ppapi_browser/ppb_file_io/ppapi_ppb_file_io.cc
// Copyright (c) 2011 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "native_client/src/shared/platform/nacl_check.h"
#include "ppapi/native_client/tests/ppapi_browser/ppb_file_io/common.h"
#include "ppapi/native_client/tests/ppapi_test_lib/get_browser_interface.h"
#include "ppapi/native_client/tests/ppapi_test_lib/test_interface.h"

// Open tests
extern void TestOpenExistingFileLocalTemporary();
extern void TestOpenNonExistingFileLocalTemporary();
// Query tests
extern void TestQueryFileLocalTemporary();
// Read tests
extern void TestCompleteReadLocalTemporary();
extern void TestParallelReadLocalTemporary();
extern void TestPartialFileReadLocalTemporary();

void SetupTests() {
  // Open tests
  RegisterTest("TestOpenExistingFileLocalTemporary",
               TestOpenExistingFileLocalTemporary);
  RegisterTest("TestOpenNonExistingFileLocalTemporary",
               TestOpenNonExistingFileLocalTemporary);
  // Query tests
  RegisterTest("TestQueryFileLocalTemporary",
               TestQueryFileLocalTemporary);
  // Read tests
  RegisterTest("TestCompleteReadLocalTemporary",
               TestCompleteReadLocalTemporary);
  RegisterTest("TestParallelReadLocalTemporary",
               TestParallelReadLocalTemporary);
  RegisterTest("TestPartialFileReadLocalTemporary",
               TestPartialFileReadLocalTemporary);
}

void SetupPluginInterfaces() {
  /* No PPP interface to test. */
}

File: ppapi/native_client/tests/ppapi_browser/ppb_file_io/ppapi_ppb_file_io.js
// Copyright (c) 2011 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

function setupTests(tester, plugin) {
  function addTest(testName, responses) {
    var expectedMessages = [testName + ':PASSED'];
    if (responses != undefined)
      expectedMessages = expectedMessages.concat(responses);
    tester.addAsyncTest('PPB_FileIO::' + testName, function(test) {
      test.expectMessageSequence(plugin, expectedMessages);
      plugin.postMessage(testName);
    });
  }
  var setupCallbacks = [
      'OpenFileSystemForSetupCallback',
      'OpenFileForSetupCallback',
      'WriteFileForSetupCallback',
      'FlushFileForSetupCallback',
      'TouchFileForSetupCallback'];
  addTest('TestOpenExistingFileLocalTemporary',
          setupCallbacks.concat(['TestOpenForRead',
                                 'TestOpenForWrite',
                                 'TestOpenTruncate',
                                 'TestOpenForWriteCreateExclusive',
                                 'TestOpenForWriteCreate',
                                 'END']));
  addTest('TestOpenNonExistingFileLocalTemporary',
          setupCallbacks.concat(['TestOpenForRead',
                                 'TestOpenForWrite',
                                 'TestOpenForWriteCreate',
                                 'DeleteFile',
                                 'TestOpenForWriteCreateExclusive',
                                 'END']));
  addTest('TestQueryFileLocalTemporary',
          setupCallbacks.concat(['OpenFileForTest',
                                 'TestQuery',
                                 'TestQueryFileVerify',
                                 'END']));
  addTest('TestPartialFileReadLocalTemporary',
          setupCallbacks.concat(['OpenFileForTest',
                                 'TestFileRead',
                                 'ReadCallback:VERIFIED',
                                 'END']));
  addTest('TestCompleteReadLocalTemporary',
          setupCallbacks.concat(['OpenFileForTest',
                                 'TestFileRead',
                                 'ReadCallback:VERIFIED',
                                 'END']));
  addTest('TestParallelReadLocalTemporary',
          setupCallbacks.concat(['OpenFileForTest',
                                 'TestParallelRead',
                                 'ReadCallback:VERIFIED',
                                 'ReadCallback:VERIFIED',
                                 'END']));
};
